### PR TITLE
Remove python_cmake_module and set hints

### PR DESCRIPTION
**Pull request by clalancette**
This must be merged before https://github.com/ros2/ros2/pull/1524 ; see that pull request for more information about this change.

### CODE REVIEW

**Review by quarkytale — APPROVED** [review]

**Comment by quarkytale** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20326/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14737/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21070/)

**Comment by clalancette** [comment]
We should hold off on merging this one until we run CI on Windows Debug.  I think there may be problems with that until we merge in https://github.com/ros2/ci/pull/755

**Comment by clalancette** [comment]
CI for this (along with all of the rest of the changes needed) is in https://github.com/ros2/ros2/pull/1524#issuecomment-2365359588

**Pull request merged**
